streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -744,7 +744,9 @@ public void run() {
     private void runLoop() {
         subscribeConsumer();
 
-        while (isRunning()) {
+        // if the thread is still in the middle of a rebalance, we should keep polling
+        // until the rebalance is completed before we close and commit the tasks
+        while (isRunning() || taskManager.isRebalanceInProgress()) {
```

Comment:
This is the actual fix.

Reply:
We need to avoid adding these new records to the `PartitionGroup` down on line 825 or else they'll be included in the offsets committed

Reply:
If we need to poll more than once to complete the rebalance for example